File: tests/benchmark/corpus/typescript/safe/safe_cross_function_dotdot.ts
// ts-safe-016: cross-function bool-returning validator with rejection.
import * as fs from 'fs';

function validateNoDotdot(s: string): boolean {
  return !s.includes('..') && !s.startsWith('/') && !s.startsWith('\\');
}

export default function (req: { query: { path: string } }) {
  const raw = req.query.path;
  if (!validateNoDotdot(raw)) return;
  return fs.readFileSync(raw);
}

File: tests/benchmark/corpus/go/safe/safe_cross_function_dotdot.go
// go-safe-016: cross-function bool-returning validator with rejection.
package main

import (
	"io/ioutil"
	"net/http"
	"strings"
)

func validateNoDotdot(s string) bool {
	return !strings.Contains(s, "..") && !strings.HasPrefix(s, "/") && !strings.HasPrefix(s, "\\")
}

func handler(w http.ResponseWriter, r *http.Request) {
	raw := r.URL.Query().Get("path")
	if !validateNoDotdot(raw) {
		return
	}
	_, _ = ioutil.ReadFile(raw)
}
